src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -399,6 +412,36 @@ private SelectItem GenerateExpandItem(ExpandTermToken tokenIn)
             List<ODataPathSegment> parsedPath = new List<ODataPathSegment>(this.parsedSegments);
             parsedPath.AddRange(pathSoFar);
 
+            // Replace the currentNavProp with a Navigation property with the derived type.
+            if (derivedType != null)

```

Comment:
>if (derivedType != null) [](http://example.com/codeflow?start=12&length=24)

It doesn't make sense to me.

Why don't we create the type case segment?

Reply:
I have update the code to create a type segment